examples/sync/src/bin/server.rs

```diff
@@ -0,0 +1,438 @@
+        let message_data = match read_message(&mut stream).await {
+            Ok(data) => data,
+            Err(NetworkError::ReadFailed(_)) => {
+                info!(client_addr = %client_addr, "Client disconnected");
+                break;
+            }
+            Err(e) => {
+                error!(client_addr = %client_addr, error = %e, "❌ Connection error");
+                state.inc_errors();
+                break;
+            }
+        };
+
+        // Parse the message
+        let message: Message = match Message::decode(&message_data[..]) {
+            Ok(msg) => msg,
+            Err(e) => {
+                error!(client_addr = %client_addr, error = %e, "❌ Failed to parse message");
+                state.inc_errors();
+                continue;
+            }
+        };
+
+        // Handle the message
+        let response = match message {
+            Message::GetOperationsRequest(request) => {
+                match handle_get_operations_request(&state, request).await {
+                    Ok(response) => Message::GetOperationsResponse(response),
+                    Err(e) => {
+                        warn!(client_addr = %client_addr, error = %e, "❌ GetOperations failed");
+                        state.inc_errors();
+                        Message::Error(e.into())
+                    }
+                }
+            }
+            Message::GetServerMetadataRequest(request) => {
+                match handle_get_server_metadata_request(&state, request).await {
+                    Ok(response) => Message::GetServerMetadataResponse(response),
+                    Err(e) => {
+                        warn!(client_addr = %client_addr, error = %e, "❌ GetServerMetadata failed");
+                        state.inc_errors();
+                        Message::Error(e.into())
+                    }
+                }
+            }
+            _ => {
+                warn!(client_addr = %client_addr, "❌ Unexpected message type");
+                state.inc_errors();
+                Message::Error(ErrorResponse::new(
+                    None,
+                    commonware_sync::ErrorCode::InvalidRequest,
+                    "Unexpected message type".to_string(),
+                ))
+            }
+        };
+
+        // Send the response with length prefix
+        let response_data = response.encode().to_vec();
+        if let Err(e) = send_message(&mut sink, &response_data).await {
+            error!(client_addr = %client_addr, error = %e, "❌ Failed to send response");
+            state.inc_errors();
+            break;
+        }
+    }
+
+    Ok(())
+}
+
+fn main() {
+    // Initialize tracing with a clean format
+    tracing_subscriber::fmt()
+        .with_target(false)
+        .with_thread_ids(false)
+        .with_file(false)
+        .with_line_number(false)
+        .init();
+
+    // Parse command line arguments
+    let matches = Command::new("ADB Sync Server")
+        .version("1.0")
```

Comment:
If you want to include version, populate it like this automatically: https://github.com/commonwarexyz/monorepo/blob/58aa05b738986567b43507243ef2451feca27eab/deployer/src/main.rs#L9-L12